vmm/src/config.rs

```diff
@@ -0,0 +1,252 @@
+            .unwrap_or_else(String::new);
+        let mut args = Cmdline::new(arch::CMDLINE_MAX_SIZE);
+        args.insert_str(cmdline_str).unwrap();
+
+        Ok(CmdlineCfg {
+            args,
+            offset: CMDLINE_OFFSET,
+        })
+    }
+    pub fn args(&self) -> &Cmdline {
+        &self.args
+    }
+    pub fn offset(&self) -> GuestAddress {
+        self.offset
+    }
+}
+
+#[derive(Debug)]
+pub struct DiskCfg<'a> {
+    path: &'a Path,
+}
+
+impl<'a> DiskCfg<'a> {
+    pub fn new(disk: &'a str) -> Result<Self> {
+        Ok(DiskCfg {
+            path: Path::new(disk),
+        })
+    }
+    pub fn path(&self) -> &Path {
+        self.path
+    }
+}
+
+pub struct RngCfg<'a> {
+    src: &'a Path,
+}
+
+impl<'a> RngCfg<'a> {
+    pub fn new(rng: Option<&'a str>) -> Result<Option<Self>> {
+        if rng.is_none() {
+            return Ok(None);
+        }
+
+        Ok(Some(RngCfg {
+            src: Path::new(rng.unwrap()),
+        }))
+    }
+    pub fn src(&self) -> &Path {
+        self.src
+    }
+}
+
+pub struct NetCfg<'a> {
+    tap: Option<&'a str>,
+    ip: Ipv4Addr,
+    mask: Ipv4Addr,
+    mac: MacAddr,
+}
+
+impl<'a> NetCfg<'a> {
+    pub fn new(net: Option<&'a str>) -> Result<Option<Self>> {
+        if net.is_none() {
+            return Ok(None);
+        }
+
+        // Split the parameters based on the comma delimiter
+        let params_list: Vec<&str> = net.unwrap().split(',').collect();
+
+        let mut tap_str: &str = "";
+        let mut ip_str: &str = "";
+        let mut mask_str: &str = "";
+        let mut mac_str: &str = "";
+
+        for param in params_list.iter() {
+            if param.starts_with("tap=") {
+                tap_str = &param[4..];
+            } else if param.starts_with("ip=") {
+                ip_str = &param[3..];
+            } else if param.starts_with("mask=") {
+                mask_str = &param[5..];
+            } else if param.starts_with("mac=") {
+                mac_str = &param[4..];
+            }
+        }
+
+        let mut tap: Option<&str> = None;
+        let mut ip: Ipv4Addr = "192.168.249.1".parse().unwrap();
+        let mut mask: Ipv4Addr = "255.255.255.0".parse().unwrap();
+        let mut mac: MacAddr = MacAddr::local_random();
+
+        if !tap_str.is_empty() {
+            tap = Some(tap_str);
+        }
+        if !ip_str.is_empty() {
+            ip = ip_str.parse().unwrap();
+        }
+        if !mask_str.is_empty() {
+            mask = mask_str.parse().unwrap();
+        }
+        if !mac_str.is_empty() {
+            mac = MacAddr::parse_str(mac_str).unwrap();
+        }
+
+        Ok(Some(NetCfg { tap, ip, mask, mac }))
+    }
+    pub fn tap(&self) -> Option<&str> {
+        self.tap
+    }
+    pub fn ip(&self) -> Ipv4Addr {
+        self.ip
+    }
+    pub fn mask(&self) -> Ipv4Addr {
+        self.mask
+    }
+    pub fn mac(&self) -> MacAddr {
+        self.mac
+    }
+}
+
+pub struct VmCfg<'a> {
```

Comment:
nit: Can we call that one `VmConfig` ? And be consistent with having `Config` instead of `Cfg`?

Reply:
Agree